cerebrium/cortex/advanced-functionality/faster-model-loading.mdx

```diff
@@ -0,0 +1,282 @@
+
+In this section below, we'll show you how to use **Tensorizer** to load your model from storage straight into GPU memory in a single step.
+
+### Installation
+
+Add the following to your `requirements.txt` file to install **Tensorizer** in your deployment:
+
+```txt
+tensorizer
+```
+
+### Usage
+
+To use **Tensorizer**, you need to first serialise your model and save it to your persistent-storage.
+
+```python
+from tensorizer import TensorSerializer
+def serialise_model(model, model_path):
+    """Serialise the model and saving the weights to the model_path"""
+    try:
+        serializer = TensorSerializer(model_path)
+        start = time.time()
+        serializer.write_module(model)
+        end = time.time()
+        print((f"Serialising model took {end - start} seconds"),  file=sys.stderr)
+        serializer.close()
+        return True
+    except Exception as e:
+        print("Serialisation failed with error: ", e,  file=sys.stderr)
+        return False
+```
+
+This will convert your model to a protocol buffer serialised format that is optimised for faster transfer speeds and fast loading into GPU memory.
+
+Then, the next time your deployment starts, you can load your serialised model from storage into GPU memory in a single step.
+You would do this as follows:
+
+```python
+
+from tensorizer import TensorDeserializer
+from tensorizer.utils import no_init_or_tensor
+def deserialise_saved_model(model_path, model_id, plaid=True):
+    """Deserialise the model from the model_path and load into GPU memory"""
+
+    # create a config object that we can use to init an empty model
+    config = AutoConfig.from_pretrained(model_id)
+
+    # Init an empty model without loading weights into gpu. We'll load later.
+    print(("Initialising empty model"),  file=sys.stderr)
+    start = time.time()
+    with no_init_or_tensor():
+        # Load your model here using whatever class you need to initialise an empty model from a config.
+        model = AutoModelForCausalLM.from_config(config)
+    end_init = time.time() - start
+
+    # Create the deserialiser object
+    #   Note: plaid_mode is a flag that does a much faster deserialisation but isn't safe for training.
+    #    -> only use it for inference.
+    deserializer = TensorDeserializer(model_path, plaid_mode=True)
+
+    # Deserialise the model straight into GPU (zero-copy)
+    print(("Loading model"),  file=sys.stderr)
+    start = time.time()
+    deserializer.load_into_module(model)
+    end = time.time()
+    deserializer.close()
+
+    # Report on the timings.
+    print(f"Initialising empty model took {end_init} seconds",  file=sys.stderr)
+    print((f"\nDeserialising model took {end - start} seconds\n"),  file=sys.stderr)
+
+    return model
+```
+
+Placing this all together in a script that you could use to load your model would look like this.
```

Comment:
I would rather put an examle of the entire script/example in a github repo and say "If you would like to see what this looks like all together look at the github repo here"

Reply:
Sounds good.
Should we leave it here until we redo the examples?
Or take it out now? @milo157